coldfront/plugins/qumulo/tests/test_tasks.py

```diff
@@ -245,3 +261,256 @@ def test_conditionally_update_storage_allocation_statuses_checks_all_pending_all
+        page_limit = qumulo_api.QumuloAPI.get_result_set_page_limit()
+        self.assertIsNotNone(page_limit)
+
+    @skip("Until we have a chance to propagte the ENV variable.")
+    def test_qumulo_result_set_page_limit_should_raise_an_exception_if_not_set(
+        self,
+    ) -> None:
+        old_page_limit = os.environ.get("QUMULO_RESULT_SET_PAGE_LIMIT")
+        os.environ["QUMULO_RESULT_SET_PAGE_LIMIT"] = ""
+
+        with self.assertRaises(TypeError):
+            qumulo_api.QumuloAPI.get_result_set_page_limit()
+
+        os.environ["QUMULO_RESULT_SET_PAGE_LIMIT"] = old_page_limit
+
+    def test_after_allocation_create_usage_is_zero(self) -> None:
+        for path in get_mock_quota_data(self.STORAGE2_PATH).keys():
+            allocation_attribute_usage = None
+            try:
+                storage_filesystem_path_attribute = AllocationAttribute.objects.select_related(
+                    "allocation"
+                ).get(
+                    value=path.rstrip("/"),
+                    allocation_attribute_type=self.storage_filesystem_path_attribute_type,
+                    allocation__status=self.status_active,
+                )
+                allocation = storage_filesystem_path_attribute.allocation
+
+                storage_quota_attribute_type = AllocationAttribute.objects.get(
+                    allocation=allocation,
+                    allocation_attribute_type=self.storage_quota_attribute_type,
+                )
+                allocation_attribute_usage = (
+                    storage_quota_attribute_type.allocationattributeusage
+                )
+            except AllocationAttribute.DoesNotExist:
+                # When the storage_path_attribute for path is not found,
+                # the allocation_attribute_usage should not exist.
+                self.assertIsNone(allocation_attribute_usage)
+                continue
+
+            self.assertEqual(allocation_attribute_usage.value, 0)
+            self.assertEqual(allocation_attribute_usage.history.first().value, 0)
+            self.assertEqual(allocation_attribute_usage.history.count(), 1)
+
+    @patch("coldfront.plugins.qumulo.tasks.QumuloAPI")
+    def test_after_getting_daily_usages_from_qumulo_api(
+        self, qumulo_api_mock: MagicMock
+    ) -> None:
+        qumulo_api = MagicMock()
+        qumulo_api.get_all_quotas_with_usage.return_value = self.mock_quota_response
+        qumulo_api_mock.return_value = qumulo_api
+
+        try:
+            ingest_quotas_with_daily_usage()
+        except:
+            self.fail("Ingest quotas raised exception")
+
+        base_quotas = get_mock_quota_base_allocations(self.STORAGE2_PATH)
+        for path, quoata_data in base_quotas.items():
+            storage_filesystem_path_attribute = AllocationAttribute.objects.select_related(
+                "allocation"
+            ).get(
+                value=path.rstrip("/"),
```

Comment:
@harterjwustl It is still possible to save an allocation with a trailing slash. This should work as long as the test data has paths without trailing slashes.